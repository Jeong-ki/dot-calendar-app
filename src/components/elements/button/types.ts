export interface IButtonProps {
  onClick: () => void;
  disabled?: boolean;
  isLoading?: boolean;
  width?: number;
  text?: string;
}
