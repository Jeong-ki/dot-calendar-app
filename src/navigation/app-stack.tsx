import React from 'react';
import type {StackParamList} from './types';
import {HomeScreen, UserScreen} from '@/screens';
import {RouteNames} from './route-names';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

const Tab = createNativeStackNavigator<StackParamList>();

export default function AppStack() {
  return (
    <Tab.Navigator initialRouteName={RouteNames.home}>
      <Tab.Screen
        name={RouteNames.home}
        component={HomeScreen as React.ComponentType}
        options={{headerShown: false}}
      />
      <Tab.Screen
        name={RouteNames.user}
        component={UserScreen as React.ComponentType}
        options={{headerShown: false}}
      />
    </Tab.Navigator>
  );
}
