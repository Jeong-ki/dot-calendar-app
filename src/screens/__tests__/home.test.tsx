import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { RouteNames } from '@/navigation/route-names';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { StackParamList } from '@/navigation/types';
import { RouteProp } from '@react-navigation/native';
import HomeScreen from '../home';

type NavigationProp = NativeStackNavigationProp<StackParamList, RouteNames.home>;
type RoutePropType = RouteProp<StackParamList, RouteNames.home>;

const mockNavigate = jest.fn();

// navigation과 route를 모킹
const mockNavigation: Partial<NavigationProp> = {
  navigate: mockNavigate,
};

const mockRoute: RoutePropType = {
  key: 'home-key',
  name: RouteNames.home,
  params: undefined, // 필요한 경우 route params 추가
};

const component = <HomeScreen navigation={mockNavigation as NavigationProp} route={mockRoute} />;

describe('HomeScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders without crashing', () => {
    const rendered = render(component);
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  it('should update testValue when "First" button is clicked', () => {
    const { getByText } = render(component);

    fireEvent.press(getByText('First'));
    expect(getByText('First Value')).toBeTruthy();
  });

  it('should update testValue when "Second" button is clicked', () => {
    const { getByText } = render(component);

    fireEvent.press(getByText('Second'));
    expect(getByText('Second Value')).toBeTruthy();
  });

  it('should navigate to user screen when "Go User" button is clicked', () => {
    const { getByText } = render(component);

    for (let i = 0; i < 3; i++) {
      fireEvent.press(getByText('Go User'));
    }
    
    expect(mockNavigate).toHaveBeenCalledTimes(3);
    expect(mockNavigate).toHaveBeenCalledWith(RouteNames.user);
  });
});
