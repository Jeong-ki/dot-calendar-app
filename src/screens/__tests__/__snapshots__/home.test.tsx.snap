// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders without crashing 1`] = `
<View>
  <Text>
    Home
  </Text>
  <Text>
    Initial Value
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#42a5f5",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 5,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        {
          "width": 100,
        },
        false,
      ]
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      First
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#42a5f5",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 5,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        {
          "width": 100,
        },
        false,
      ]
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      Second
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#42a5f5",
          "borderRadius": 5,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 5,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        {
          "width": 100,
        },
        false,
      ]
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      Go User
    </Text>
  </View>
</View>
`;
